raiden-network/light-client
# Refactor errors to JSON and use explicit validation errors

Part of #173 

**Short description**
Needed so the CLI refactoring (#1636) can rely on SDK's validations, we need to have explicit error codes/messages thrown upon such validation errors.
The catched errors from API methods can be compared either by message or by code literal, like:
```ts
try {
  await raiden.getAvailability('0xinvalid');
} catch (err) {
  err.message === ErrorCodes.DTA_INVALID_ADDRESS; // by message
  err.code === 'DTA_INVALID_ADDRESS'; // literal/narrowed type, may require checking instanceof RaidenError
}
```

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Try the approach on example, and check if it works
2. Rest of API is tested by e2e tests
